app/src/main/java/br/com/dionataferraz/vendas/transaction/TransactionsActivity.kt

```diff
@@ -1,77 +1,103 @@
         setContentView(binding.root)
         configureActionBar()
 
-        binding.rcList.adapter = adapter
-
-        adapter.addList(getTransactionList())
-//        setupTransactionList()?.let { adapter.addList(it) }
-    }
-
-    @RequiresApi(Build.VERSION_CODES.O)
-    private fun setupTransactionList(): List<TransactionEntity>? {
-        val sharedPreferences = getSharedPreferences("transacListPrefs", Context.MODE_PRIVATE);
-        var jsonTransactions = sharedPreferences.getString("transactionList", "");
-        val jsonAdapter = getJsonTransactionListAdapter();
-
-        if (jsonTransactions.isNullOrBlank()) {
-            val listJson = jsonAdapter?.toJson(getTransactionList())
-            sharedPreferences
-                .edit()
-                .putString("transactionList", listJson)
-                .apply()
-
-            jsonTransactions = sharedPreferences.getString("transactionList", "")
+        viewModel = TransactionViewModel()
+        getProgressBar().visibility = View.VISIBLE
+        viewModel.updateTrasactionView()
+
+        val itemTouchHelper = ItemTouchHelper(getSwipe())
+        itemTouchHelper.attachToRecyclerView(binding.rcList)
+
+        viewModel.transactionList.observe(this) { transactions ->
+            if (transactions != null && !transactions.isEmpty()) {
+                val transactionsE = transactions.stream().map {
+                    TransactionEntity(
+                        id = it.id,
+                        amount = it.value,
+                        name = it.description,
+                        time = it.transactionDate,
+                        transactionType = it.transactionType
+                    )
+                }.toList()
+                binding.rcList.adapter = adapter
+                adapter.addList(transactionsE)
+            } else {
+                adapter.clear()
+            }
+            getProgressBar().visibility = View.INVISIBLE
+            getItensView().visibility = View.VISIBLE
         }
+    }
 
-        val listTransactionJson = jsonTransactions?.let { jsonAdapter?.fromJson(it) };
-
-        return listTransactionJson
+    fun getProgressBar(): ProgressBar {
+        return binding.transactionsProgressBar
     }
 
-    fun getJsonTransactionListAdapter(): JsonAdapter<List<TransactionEntity>>? {
-        val moshi = Moshi
-            .Builder()
-            .add(Date::class.java, Rfc3339DateJsonAdapter().nullSafe())
-            .addLast(KotlinJsonAdapterFactory())
-            .build()
-        val listType = Types.newParameterizedType(List::class.java, TransactionEntity::class.java)
+    fun getItensView(): RecyclerView {
+        return binding.rcList
+    }
 
-        return moshi.adapter(listType)
+    fun getSwipe(): ItemTouchHelper.SimpleCallback {
+        return object : ItemTouchHelper.SimpleCallback(
+            0,
+            ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT or ItemTouchHelper.DOWN or ItemTouchHelper.UP
```

Comment:
Deixou de ser um swipe pois foi definido que podemos arrastar para qualquer lado para fazer a remoção
```suggestion
    fun getDrag(): ItemTouchHelper.SimpleCallback {
        return object : ItemTouchHelper.SimpleCallback(
            0,
            ItemTouchHelper.LEFT or ItemTouchHelper.RIGHT or ItemTouchHelper.DOWN or ItemTouchHelper.UP
```